### PR TITLE
Migrate from Travis-CI to Github Actions

### PR DESCRIPTION
**Pull request by leifmadsen**

### CODE REVIEW

**Comment by leifmadsen** [comment]
[Migrate from Travis-CI to GitHub Actions](https://trello.com/c/QVFdw42L/1210-migrate-from-travis-ci-to-github-actions)

File: build/test-framework/run_tests.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ sed -i '/^tsflags=.*/a ip_resolve=4' /etc/yum.conf
 yum install -y epel-release
```
Comment (mrunge) [inline build/test-framework/run_tests.sh:12]: yum should become dnf

**Review by mrunge — COMMENTED** [review]
In general, looks good to me.
One nit re. use of yum;

the other issue here I see is the use of EPEL. It would be good to get rid of that.

**Review by mrunge — CHANGES_REQUESTED** [review]
Please switch to dnf

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+        run: |
+          docker run -p 9200:9200 --name elastic -p 9300:9300 -e "discovery.type=single-node" -d docker.elastic.co/elasticsearch/elasticsearch:$ELASTIC_VERSION
+
+      - name: Start QDR
+        run: |
+          docker run -p 5672:5672 -d quay.io/interconnectedcloud/qdrouterd:$QDROUTERD_VERSION
+
+      - name: Run unit testing and code coverage
+        run: |
+          docker run -eCOVERALLS_TOKEN -uroot --network host -i --volume $GITHUB_WORKSPACE:/go/src/github.com/infrawatch/smart-gateway:z --workdir /go/src/github.com/infrawatch/smart-gateway centos:8 /bin/sh -c 'sh ./build/test-framework/run_tests.sh'
```
Comment (leifmadsen) [inline .github/workflows/main.yml:27]: @mrunge still using centos:8 here, so will need to see if I can address that. The big blocker to changing that is the AMQP library. Need to see if there is another repo I can enable.

File: build/test-framework/run_tests.sh (hunk 1 above)
Comment (leifmadsen) [reply build/test-framework/run_tests.sh:12]: Yea this is what was in here before, but I think we can make that adjustment here too.

File: .github/workflows/main.yml (hunk 2 above)
Comment (leifmadsen) [reply .github/workflows/main.yml:27]: I ended up taking a shot at migrating us to ubi8 actually. See #99 which is requested to merge here.

File: build/test-framework/run_tests.sh (hunk 1 above)
Comment (leifmadsen) [reply build/test-framework/run_tests.sh:12]: See #99

**Review by mrunge — APPROVED** [review]
thanks!

**Pull request merged**